### PR TITLE
Update Azure SDK to 1.2.27

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -225,7 +225,7 @@
             <dependency>
                 <groupId>com.azure</groupId>
                 <artifactId>azure-sdk-bom</artifactId>
-                <version>1.2.25</version>
+                <version>1.2.27</version>
```
Comment (nineinchnick) [inline pom.xml:228]: I tested it locally and it still fails in the same way that 1.2.26 did:
```
[INFO] Running io.trino.filesystem.azure.TestAzureFileSystemGen2Flat
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.657 s <<< FAILURE! -- in io.trino.filesystem.azure.TestAzureFileSystemGen2Flat
[ERROR] io.trino.filesystem.azure.TestAzureFileSystemGen2Flat -- Time elapsed: 0.657 s <<< ERROR!
java.io.IOException: Failed to check whether hierarchical namespaces is enabled for the storage account devcicdflat and container test-flat-410daf35-ff83-4886-b4c6-0b0ac087ab37
	at io.trino.filesystem.azure.AbstractTestAzureFileSystem.isHierarchicalNamespaceEnabled(AbstractTestAzureFileSystem.java:120)
	at io.trino.filesystem.azure.AbstractTestAzureFileSystem.initialize(AbstractTestAzureFileSystem.java:95)
	at io.trino.filesystem.azure.AbstractTestAzureFileSystem.initializeWithAccessKey(AbstractTestAzureFileSystem.java:69)
	at io.trino.filesystem.azure.TestAzureFileSystemGen2Flat.setup(TestAzureFileSystemGen2Flat.java:32)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1489)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2071)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2033)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 0
	at com.azure.storage.file.datalake.DataLakePathAsyncClient.<init>(DataLakePathAsyncClient.java:165)
	at com.azure.storage.file.datalake.DataLakeDirectoryAsyncClient.<init>(DataLakeDirectoryAsyncClient.java:89)
	at com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient.getDirectoryAsyncClient(DataLakeFileSystemAsyncClient.java:211)
	at com.azure.storage.file.datalake.DataLakeFileSystemClient.getDirectoryClient(DataLakeFileSystemClient.java:191)
	at io.trino.filesystem.azure.AbstractTestAzureFileSystem.isHierarchicalNamespaceEnabled(AbstractTestAzureFileSystem.java:117)
	... 10 more
```

**Comment by wendigo** [comment]
@nineinchnick the SDK now has a dedicated method to check account settings that we can use to check whether filesystem is hierarchical

File: lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java [hunk 2]
```diff
@@ -445,8 +447,8 @@ private boolean isHierarchicalNamespaceEnabled(AzureLocation location)
             throws IOException
     {
         try {
-            DataLakeFileSystemClient fileSystemClient = createFileSystemClient(location);
-            return fileSystemClient.getDirectoryClient("/").exists();
+            BlobContainerClient client = createBlobContainerClient(location);
+            return client.getAccountInfo(Duration.of(3, SECONDS)).isHierarchicalNamespaceEnabled();
```
Comment (anusudarsan) [inline lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java:451]: how did you come up with the timeout value?

Comment (wendigo) [reply lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java:451]: It doesn't work so :)

**Pull request closed without merging**